lib/kdl/test/src/result_test.cpp

```diff
@@ -293,7 +294,7 @@ namespace kdl {
      */
     template <typename ResultType, typename E>
     void test_visit_error_const_lvalue_ref_with_opt_value(E&& e) {
-        auto result = ResultType::error(std::forward<E>(e));
+        auto result = ResultType(std::forward<E>(e));
```

Comment:
we're losing the guarantee that these are errors.
Needs a `CHECK(result.is_error());` on all of these.

Reply:
Ah, good catch. I'll add a `REQUIRE`.